src/preferences/colorpaletteeditormodel.cpp

```diff
@@ -103,9 +103,15 @@ void ColorPaletteEditorModel::setColorPalette(const ColorPalette& palette) {
     // Make a map of hotcue indices
     QMap<int, int> hotcueColorIndicesMap;
     QList<unsigned int> hotcueColorIndices = palette.getHotcueIndices();
-    for (int i = 0; i < hotcueColorIndices.size(); i++) {
-        int colorIndex = hotcueColorIndices.at(i);
-        hotcueColorIndicesMap.insert(colorIndex, i);
+    if (hotcueColorIndices.size()) {
+        for (int i = 0; i < hotcueColorIndices.size(); i++) {
+            int colorIndex = hotcueColorIndices.at(i);
+            hotcueColorIndicesMap.insert(colorIndex, i);
+        }
+    } else {
+        for (int i = 0; i < palette.size(); i++) {
+            hotcueColorIndicesMap.insert(i, i);
```

Comment:
Now it's a bit cumbersome so set a single default color for a large palette because you'd have to remove all the other numbers in the index column by hand. It's also a bit confusing because we now have numbers for the built-in track color palettes, too. The previous situation was probably also confusing, so I'm not sure which one is better. If we keep this change I think we need a context menu item to unset all indices in the palette editor.

Reply:
The idea was to have always the coloumn populated with the cue numbers and enable the use via the box. 
So you only need to edit the column if you are not fine with this. This should effect virtually no one, isn't it?

Reply:
Let's say I have a palette with 10 colors. One of the colors is red. If I want all my new cues red, I have to set "assign to hotcue number" to 1 for red and an empty value for all others. Before, I could just double click next to the red color and type 1. Done.

Now I have to the the same and then delete the numbers from all other rows.

Reply:
I think this should be move into the paletteditor(model) classes. If there are no assigned hotcue indices, we could display grey placeholder values in the "assign to hot cue numbers" table (the placeholder values are not saved) . As soon as you set one value eyplicitly, the placeholder values should disappear.

Reply:
> Before, I could just double click next to the red color and type 1. Done.

Interesting. That was not self explaining to me. I have actually tried to add all cue numbers to red like 1, 2, 3 ... but it fails. 

A combobox with all palette colors + color by number for default cue color will be IMHO quite easy to understand compared to that.  

I like to avoid to bothering the user with the palette editor if he just want to select the default color.  

Does this work for you as well?

Reply:
I agree that setting the default color via the palette editor causes too much friction, however let's keep this out of this PR as I have an idea for that that I'd like to try in a separate PR after this one has been merged.

Reply:
> I think this should be move into the paletteditor(model) classes. If there are no assigned hotcue indices, we could display grey placeholder values in the "assign to hot cue numbers" table (the placeholder values are not saved) . As soon as you set one value explicitly, the placeholder values should disappear.

Can you implement this? The current change also prevent reordering the palette with no indices set. Also, palette could be big so you should check if `i` is less than the number of hotcues.